### PR TITLE
Add post processing to files generated by TS generators 

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.3.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Allow users to define the environment variable "TS_POST_PROCESS_FILE" to post process TS files generated by TS generators (e.g. Angular, Fetch, etc), e.g. `export TS_POST_PROCESS_FILE='/usr/local/bin/prettier --write'`

Please refer to 07b2a36 for the code changes done by `prettier` as an example.

cc @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -561,4 +566,30 @@ public String escapeQuotationMark(String input) {
     public String escapeUnsafeCharacters(String input) {
         return input.replace("*/", "*_/").replace("/*", "/_*");
     }
+
+    @Override
+     public void postProcessFile(File file, String fileType) {
+         if (file == null) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: i suggest to call the parent method, too, since there could be some logic added later.

File: samples/client/petstore/typescript-node/default/model/apiResponse.ts [hunk 2]
```diff
@@ -3,43 +3,46 @@
...
  * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
  * https://openapi-generator.tech
  * Do not edit the class manually.
  */
 
-
 /**
-* Describes the result of uploading an image resource
-*/
+ * Describes the result of uploading an image resource
```
Comment (macjohnny) [inline samples/client/petstore/typescript-node/default/model/apiResponse.ts]: why was this changed? the generation parameters of the `typescript-node` samples have not changed, so I guess this is because of a local test?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: Sure. I'll do it later.

File: samples/client/petstore/typescript-node/default/model/apiResponse.ts (hunk 2 above)
Comment (wing328) [reply samples/client/petstore/typescript-node/default/model/apiResponse.ts]: Restore via 9573d8e

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: Done via 9573d8e
